src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -180,7 +180,7 @@ private static bool IsControlBlock(SyntaxNode node)
             }
 
             // * { - in the Lambda context
-            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentTokenParentParent != null &&
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) &&
                (currentTokenParentParent.IsKind(SyntaxKind.SimpleLambdaExpression) || currentTokenParentParent.IsKind(SyntaxKind.ParenthesizedLambdaExpression)))
```

Comment:
Do we also want to apply this simplification?

```suggestion
               currentTokenParentParent.IsKind(SyntaxKind.SimpleLambdaExpression, SyntaxKind.ParenthesizedLambdaExpression))
```

Reply:
If this PR closes #53218 I think it should. There are a lot of similar instances to be simplified in this file.